camel/toolkits/linkedin_toolkit.py

```diff
@@ -0,0 +1,288 @@
+            "X-Restli-Protocol-Version": "2.0.0",
+        }
+
+        response = requests.get(
+            "https://api.linkedin.com/v2/userinfo",
+            headers=headers,
+        )
+
+        if response.status_code != HTTPStatus.OK:
+            return ""
+
+        json_response = response.json()
+        user_urn = f"urn:li:person:{json_response['sub']}"
+        return user_urn
+
+    def get_tools(self) -> List[OpenAIFunction]:
+        r"""Returns a list of OpenAIFunction objects representing the
+        functions in the toolkit.
+
+        Returns:
+            List[OpenAIFunction]: A list of OpenAIFunction objects
+                representing the functions in the toolkit.
+        """
+        return [
+            OpenAIFunction(self.create_post),
+            OpenAIFunction(self.delete_post),
+            OpenAIFunction(self.get_my_profile),
+        ]
+
+    def _get_access_token(self) -> str:
+        r"""Fetches the access token required for making LinkedIn API requests.
+
+        Returns:
+            str: The OAuth 2.0 access token.
+
+        Note:
+            This function assumes the existence of an environment variable
+            `LINKEDIN_ACCESS_TOKEN` which stores the access token.
+        """
+        return os.getenv("LINKEDIN_ACCESS_TOKEN", "")
+
+    def _handle_http_error(self, response: requests.Response) -> str:
+        r"""Handles the HTTP errors based on the status code of the response.
+
+        Args:
+            response (requests.Response): The HTTP response from the API call.
+
+        Returns:
+            str: The error type, based on the status code.
+        """
+        if response.status_code == HTTPStatus.UNAUTHORIZED:
+            return "Unauthorized. Check your access token."
+        elif response.status_code == HTTPStatus.FORBIDDEN:
+            return (
+                "Forbidden. You do not have permission to perform this action."
+            )
+        elif response.status_code == HTTPStatus.NOT_FOUND:
+            return "Not Found. The resource could not be located."
+        elif response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+            return "Too Many Requests. You have hit the rate limit."
+        else:
+            return "HTTP Error"
+
+
+LINKEDIN_FUNCS: List[OpenAIFunction] = LinkedInToolkit().get_tools()
+
+
+if __name__ == "__main__":
+    toolkit = LinkedInToolkit()
+
+    # test to get user profile
+    print("Testing get_my_profile:")
+    profile_info = toolkit.get_my_profile()
+    print(profile_info)
+
+    # test to create a post
+    print("\nTesting create_post:")
+    post_text = "This is a test post created by LinkedInToolkit @Camel."
+    create_result = toolkit.create_post(text=post_text)
+    print(create_result)
+
+    # test to delete a post(need a valid post_id)
+    print("\nTesting delete_post:")
+    post_id = create_result['Post ID']
+    delete_result = toolkit.delete_post(post_id=post_id)
+    print(delete_result)
```

Comment:
please move the test code to camel/test/toolkits

Reply:
removed